### ISSUE TITLE
Check failed: *ptr host allocation of size 2167603200 failed

### ISSUE THREAD

**Issue by yonglinZ** [alexgkendall/SegNet-Tutorial#81]
When I input the command below（cpu mode）：
python Scripts/test_segmentation_camvid.py --model Example_Models/segnet_basic_camvid.prototxt --weights Example_Models/segnet_basic_camvid.caffemodel --iter 23

Error：
I0221 05:51:30.850589 17851 net.cpp:410] dense_softmax_inner_prod <- conv_decode1
I0221 05:51:30.850597 17851 net.cpp:368] dense_softmax_inner_prod -> dense_softmax_inner_prod
I0221 05:51:30.850606 17851 net.cpp:120] Setting up dense_softmax_inner_prod
I0221 05:51:30.850870 17851 net.cpp:127] Top shape: 1 11 360 480 (1900800)
I0221 05:51:30.850914 17851 layer_factory.hpp:74] Creating layer argmax
I0221 05:51:30.850924 17851 net.cpp:90] Creating Layer argmax
I0221 05:51:30.850929 17851 net.cpp:410] argmax <- dense_softmax_inner_prod
I0221 05:51:30.850935 17851 net.cpp:368] argmax -> argmax
I0221 05:51:30.850941 17851 net.cpp:120] Setting up argmax
I0221 05:51:30.850951 17851 net.cpp:127] Top shape: 1 1 360 480 (172800)
I0221 05:51:30.850956 17851 net.cpp:194] argmax does not need backward computation.
I0221 05:51:30.850958 17851 net.cpp:194] dense_softmax_inner_prod does not need backward computation.
I0221 05:51:30.850962 17851 net.cpp:194] conv_decode1_bn does not need backward computation.
I0221 05:51:30.850965 17851 net.cpp:194] conv_decode1 does not need backward computation.
I0221 05:51:30.850970 17851 net.cpp:194] upsample1 does not need backward computation.
I0221 05:51:30.850975 17851 net.cpp:194] conv_decode2_bn does not need backward computation.
I0221 05:51:30.850986 17851 net.cpp:194] conv_decode2 does not need backward computation.
I0221 05:51:30.850991 17851 net.cpp:194] upsample2 does not need backward computation.
I0221 05:51:30.850997 17851 net.cpp:194] conv_decode3_bn does not need backward computation.
I0221 05:51:30.851004 17851 net.cpp:194] conv_decode3 does not need backward computation.
I0221 05:51:30.851008 17851 net.cpp:194] upsample3 does not need backward computation.
I0221 05:51:30.851013 17851 net.cpp:194] conv_decode4_bn does not need backward computation.
I0221 05:51:30.851017 17851 net.cpp:194] conv_decode4 does not need backward computation.
I0221 05:51:30.851022 17851 net.cpp:194] upsample4 does not need backward computation.
I0221 05:51:30.851028 17851 net.cpp:194] pool4 does not need backward computation.
I0221 05:51:30.851033 17851 net.cpp:194] relu4 does not need backward computation.
I0221 05:51:30.851037 17851 net.cpp:194] conv4_bn does not need backward computation.
I0221 05:51:30.851042 17851 net.cpp:194] conv4 does not need backward computation.
I0221 05:51:30.851047 17851 net.cpp:194] pool3 does not need backward computation.
I0221 05:51:30.851052 17851 net.cpp:194] relu3 does not need backward computation.
I0221 05:51:30.851058 17851 net.cpp:194] conv3_bn does not need backward computation.
I0221 05:51:30.851063 17851 net.cpp:194] conv3 does not need backward computation.
I0221 05:51:30.851068 17851 net.cpp:194] pool2 does not need backward computation.
I0221 05:51:30.851073 17851 net.cpp:194] relu2 does not need backward computation.
I0221 05:51:30.851078 17851 net.cpp:194] conv2_bn does not need backward computation.
I0221 05:51:30.851083 17851 net.cpp:194] conv2 does not need backward computation.
I0221 05:51:30.851088 17851 net.cpp:194] pool1 does not need backward computation.
I0221 05:51:30.851091 17851 net.cpp:194] relu1 does not need backward computation.
I0221 05:51:30.851096 17851 net.cpp:194] conv1_bn does not need backward computation.
I0221 05:51:30.851100 17851 net.cpp:194] conv1 does not need backward computation.
I0221 05:51:30.851106 17851 net.cpp:194] norm does not need backward computation.
I0221 05:51:30.851111 17851 net.cpp:235] This network produces output argmax
I0221 05:51:30.851131 17851 net.cpp:482] Collecting Learning Rate and Weight Decay.
I0221 05:51:30.851177 17851 net.cpp:247] Network initialization done.
I0221 05:51:30.851184 17851 net.cpp:248] Memory required for data: 392263680
F0221 05:51:36.485659 17851 syncedmem.hpp:27] Check failed: *ptr host allocation of size 2167603200 failed
*** Check failure stack trace: ***
Aborted (core dumped)

Please help me about this error. THX

**Comment by WeiyiLi** [comment]
I guess you have run out of your memory on your CPU.
